### DISCUSSION TITLE
Principal stress is used in a material model

### DISCUSSION THREAD

**Discussion by PEI0214** [idaholab/moose#28994] [Q&A General]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

Hello，
I want to add a material model. The elastic modulus of this model is related to the principal stress. When I use meshmodifiers to simulate construction, the process is shown below. 
![image](https://github.com/user-attachments/assets/1b1fe279-1918-4daf-9c86-0b50b987c3f0)

When the type of principal stress is nodal variable, the inactive part also has principal stress. When the principal stress type is elemental variable, the principal stress value of the inactive part is 0, as shown in the figure below.
![image](https://github.com/user-attachments/assets/415a2eb8-86dd-412d-91dc-c8a680119cba)

However, the solving type of the principal stress in moose is node type, which will affect my solving of the elastic modulus value. Is there a way to use element type principal stress in a material model?
minPrincipal in MOOSE:
~~~
template <typename T>
T
minPrincipal(const RankTwoTensorTempl<T> & r2tensor, Point & direction)
{
  return calcEigenValuesEigenVectors(r2tensor, 0, direction);
}
~~~

**Comment by GiudGiud** [comment]
Hello

We have not implemented any default behavior for material properties on newly activated subdomains. 
All the work has focused on variables so far. If you want a particular behavior you will need to implement it

For the figure you show, that's a visualization artefact. That's just how paraview plot nodal variables, as continuous variables by default

Guillaume

**Reply by lynnmunday** [reply]
I don't understand what you are wanting to implement but stress is a material property defined at the gauss points, not the nodes. 
 You can compute a "nodal stress" as a post processing step from the stress material property but that "nodal stress" isn't the stress used in a constitutive model.  ComputeSmearedCrackingStress is a material that uses principal strains to modify material properties.  Maybe you can do something similar.
